src/lib/interceptor/search-request.interceptor.spec.ts

```diff
@@ -207,6 +209,19 @@ describe('SearchRequestInterceptor', () => {
                 withDeleted: false,
             });
         });
+
+        it('should throw when operand type not equals entity field type', async () => {
+            const invalidWhereList = [
+                { where: [{ col1: { operator: 'BETWEEN', operand: [1, 2] } }] },
+                { where: [{ col1: { operator: 'IN', operand: [1] } }] },
+                { where: [{ col2: { operator: 'IN', operand: ['1'] } }] },
+                { where: [{ col1: { operator: '>', operand: 123 } }] },
+            ];
+
+            for (const where of invalidWhereList) {
+                await expect(interceptor.validateBody(where)).rejects.toThrow(UnprocessableEntityException);
+            }
+        });
```

Comment:
`col1` is defined as string, and `col2` as number.
should validate type equality of operand and entity field